File: simple/java/extract-method_isolate-switch_after.java
class Order {
// ...

double getTotal() {
  double total = 0;
  for (Product product : getProducts()) {
    total = product.quantity * product.price;
  }
  total = applyRegionalDiscounts(total);
  return total;
}

double applyRegionalDiscounts(total) {
  double result = total;
  switch (user.getCountry()) {
    'US': result *= 0.85; break;
    'RU': result *= 0.75; break;
    'CN': result *= 0.9; break;
    // ...
  }
  return result;
}

File: simple/php/extract-method_isolate-switch_after.php
class Order {
// ...

function calculateTotal() {
  $total = 0;
  foreach ($this->getProducts() as $product) {
    $total = $product->quantity * $product->price;
  }
  $total = applyRegionalDiscounts($total);
  return $total;
}

function applyRegionalDiscounts($total) {
  $result = $total;
  switch ($this->user->getCountry()) {
    'US': $result *= 0.85; break;
    'RU': $result *= 0.75; break;
    'CN': $result *= 0.9; break;
    // ...
  }
  return $result;
}